tests/test_indieauth/test_profile.py

```diff
@@ -0,0 +1,130 @@
+import pytest
+import responses
+
+from indieweb_utils.indieauth import profile
+
+
+@pytest.fixture
+def name_tag():
+    return '<h1 class="p-name">John Doe</h1>'
+
+
+@pytest.fixture
+def photo_tag():
+    return '<img src="http://example.com/me.jpg" class="u-photo"/>'
+
+
+@pytest.fixture
+def me_tag():
+    return '<a href="http://example.com/john" class="u-url"></a>'
+
+
+@pytest.fixture
+def email_tag():
+    return '<a href="mailto:john@example.com" class="u-email"></a>'
+
+
+@responses.activate
+def test_parses_profile(name_tag, photo_tag, me_tag, email_tag):
+    url = "http://example.com"
+    responses.add(
+        responses.Response(
+            method="GET",
+            url=url,
+            body=f'<div class="h-card">{name_tag}{photo_tag}{me_tag}{email_tag}</div>',
+        )
+    )
+    actual = profile.get_profile(me=url)
+    assert actual.name == "John Doe"
+    assert actual.photo == "http://example.com/me.jpg"
+    assert actual.url == "http://example.com/john"
+    assert actual.email == "john@example.com"
+
+
+@responses.activate
+def test_handles_missing_email(name_tag, photo_tag, me_tag):
+    url = "http://example.com"
+    responses.add(
+        responses.Response(
+            method="GET",
+            url=url,
+            body=f'<div class="h-card">{name_tag}{photo_tag}{me_tag}</div>',
+        )
+    )
+    actual = profile.get_profile(me=url)
+    assert actual.email is None
+
+
+@responses.activate
+def test_handles_missing_url(name_tag, photo_tag):
+    """Should use me url for url if not found."""
+    url = "http://example.com"
+    responses.add(
+        responses.Response(
+            method="GET",
+            url=url,
+            body=f'<div class="h-card">{name_tag}{photo_tag}</div>',
+        )
+    )
+    actual = profile.get_profile(me=url)
+    assert actual.url == url
+
+
+@responses.activate
+def test_handles_missing_photo(name_tag):
+    url = "http://example.com"
+    responses.add(
+        responses.Response(
+            method="GET",
+            url=url,
+            body=f'<div class="h-card">{name_tag}</div>',
+        )
+    )
+    actual = profile.get_profile(me=url)
+    assert actual.photo is None
+
+
+@responses.activate
+def test_handles_missingadasd(name_tag):
+    url = "http://example.com"
+    responses.add(
+        responses.Response(
+            method="GET",
+            url=url,
+            body='<div class="h-card"><a class="u-email"></a></div>',
+        )
+    )
+    actual = profile.get_profile(me=url)
+    assert actual.photo is None
```

Comment:
I accidentlly committed this when I was testing how it would behave if we had a malformed a tag (i.e. no href . Please delete :)